### PR TITLE
Fixed display capture on Windows

### PR DESCRIPTION
**Pull request by SteveWestbrook**
Support noticed that display capture was no longer working on Windows.  Confirmed this and also reproduced in OBS.  Issue appears to be related to hybrid Windows systems that have onboard graphics as well as a graphics card.

- Default to hardware capture - ensures display capture will work
- Configurable to fall back to original implementation, which does not specify capture method.

Unfortunately, allowing for configuration required fairly wide-ranging modifications.  However, fallback capability is important and has therefore been implemented.

### CODE REVIEW

File: libobs/obs.c [hunk 1]
```diff
@@ -263,8 +263,13 @@ static int obs_init_graphics(struct obs_video_info *ovi)
 	bool success = true;
 	int errorcode;
 
+	#ifdef EVERCAST
```
Comment (skastr0) [inline libobs/obs.c:266]: Is this the same flag as the vendor skins?

Comment (skastr0) [reply libobs/obs.c:266]: given it's a root OBS file, how do we go about modifying this? I'm 99% sure this wouldn't be compatible with upstream ie. ifdef <company/vendor> ad infinitum

File: vendor_skins/Evercast/libobs/obs.h [hunk 2]
```diff
@@ -36,6 +36,8 @@
 #include "obs-properties.h"
 #include "obs-interaction.h"
 
+#define EVERCAST
```
Comment (skastr0) [inline vendor_skins/Evercast/libobs/obs.h:39]: oh I see, we replace the header file and that'll be defined then. Can't we simply replace both header and impl files?

Comment (SteveWestbrook) [reply vendor_skins/Evercast/libobs/obs.h:39]: Yeah, I'm not terribly happy with this entire set of changes.  We need to add this single parameter, but unfortunately, it's being added to an interface, which means dirtying a whole graph of code for this one little boolean.  The reason I didn't follow our usual practice with obs.c and copy it to vendor_skins is that it's is a pretty fundamental file.  If changes happen here upstream, they are very likely to be important, we really really don't want to miss out on them.  So, for this one instance, where we are traipsing through the file on our way somewhere else, I picked the `#ifdef` as the lesser of two evils.

**Pull request closed without merging**